src/main/java/frc/robot/Robot.java

```diff
@@ -7,96 +7,99 @@
 import java.util.List;
 
 import edu.wpi.first.math.geometry.Pose2d;
-import edu.wpi.first.math.geometry.Rotation2d;
-import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
-import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import frc.robot.commands.AutoCommandInterface;
+import frc.robot.commands.NoteAuto;
+import frc.robot.subsystems.DriveTrain;
 
 public class Robot extends TimedRobot {
-  private Command m_autonomousCommand;
-
-  private RobotContainer m_robotContainer;
-
-  @Override
-  public void robotInit() {
-    m_robotContainer = new RobotContainer();
-  }
-
-  @Override
-  public void robotPeriodic() {
-    CommandScheduler.getInstance().run();
-  }
-
-  @Override
-  public void disabledInit() {}
-
-  @Override
-  public void disabledPeriodic() {}
-
-  @Override
-  public void disabledExit() {}
-
-  @Override
-  public void autonomousInit() {
-    m_autonomousCommand = m_robotContainer.getAutonomousCommand();
-
-    // set initial robot position
-    Pose2d robotStartPose;
-    if(DriverStation.getLocation().getAsInt() == 1)
-      robotStartPose = FieldConstants.START_1;
-    else if(DriverStation.getLocation().getAsInt() == 2)
-      robotStartPose = FieldConstants.START_2;
-    else
-      robotStartPose = FieldConstants.START_3;
-    
-    // flip paths when red (have to do this because PathPlanner flips a path only)
-    if(DriverStation.getAlliance().get().toString() == "Red")
-      robotStartPose = new Pose2d(FieldConstants.FIELD_LENGTH - robotStartPose.getX(), robotStartPose.getY(), Rotation2d.fromDegrees(180.0));
-      
-    m_robotContainer.getDriveTrain().setPose(robotStartPose);
-    
-    if (m_autonomousCommand != null) {
-      m_autonomousCommand.schedule();
+    private AutoCommandInterface m_autonomousCommand;
+    private SendableChooser<AutoCommandInterface> m_chosenAuto = new SendableChooser<>();
+
+    private RobotContainer m_robotContainer;
+
+    @Override
+    public void robotInit() {
+        m_robotContainer = new RobotContainer();
+        DriveTrain driveTrain = m_robotContainer.getDriveTrain();
+        // Initialize the list of available Autonomous routines
+        m_chosenAuto.setDefaultOption("Test Auto", new NoteAuto(driveTrain));
+
+        SmartDashboard.putData("Chosen Auto", m_chosenAuto);
+    }
+
+    @Override
+    public void robotPeriodic() {
+        CommandScheduler.getInstance().run();
+    }
+
+    @Override
+    public void disabledInit() {
+    }
+
+    @Override
+    public void disabledPeriodic() {
+    }
```

Comment:
I believe we need code here to set the robot pose as people are selecting the auto. Much easier to follow when debugging. I think we had this in 2022??

Reply:
That is, inside disabledPeriodic()

Reply:
what happens if the robot does not have any April tags to get its pose

Reply:
Does not matter. If the operator chooses an Auto, we should set the robot to that starting position. You do it in autoInit(), just also do it if the Auto changes.  Look in the 2023 code; basically need all of that (except Arm code).